lib/checkfunctions.cpp

```diff
@@ -467,3 +467,45 @@ void CheckFunctions::checkLibraryMatchFunctions()
                     "--check-library: There is no matching configuration for function " + functionName + "()");
     }
 }
+
+// Check for problems to compiler apply (Named) Return Value Optimization for local variable
+// Technically we have different guarantees between standard versions
+// details: https://en.cppreference.com/w/cpp/language/copy_elision
+void CheckFunctions::returnLocalStdMove()
+{
+    if(!mTokenizer->isCPP() || mSettings->standards.cpp < Standards::CPP11)
+        return;
+
+    if (!mSettings->severity.isEnabled(Severity::performance))
+        return;
+
+    const SymbolDatabase *symbolDatabase = mTokenizer->getSymbolDatabase();
+    for (const Scope *scope : symbolDatabase->functionScopes) {
+        // Expect return by-value
+        if (Function::returnsReference(scope->function, true))
+            continue;
+
+        const auto rets = Function::findReturns(scope->function);
+        for (const Token* ret : rets) {
+            const Token* retval = ret->next();
```

Comment:
This should use the AST:

```cpp
if (!Token::simpleMatch(ret->tokAt(-3), "std :: move (")
    continue;
const Token* retval = ret->astOperand2();
// NRVO
if (retval->variable() && retval->variable()->isLocal() && !retval->variable()->isVolatile())
    copyElisionError(retval);
// RVO
if (Token::Match(retval, "(|{") && !retval->isCast())
    copyElisionError(retval);
```

This will also catch more cases such as constructors for types that aren't a single token(like `A<int>` or `A::type`) and function calls.

Reply:
Great improvement, thank you!
Still I don't get why Operand**2**, need more meditation about this)

Reply:
For most cases, operand1 is for the left-hand side, and operand2 is for the right-hand side, except unary operators(like `+`, `*`, `-`, etc) which only have operand1.